boards/shared/Src/Io/Io_SharedFreqOnlyPwmInput.c

```diff
@@ -0,0 +1,133 @@
+#include "main.h"
+#include <math.h>
+#include <assert.h>
+#include "Io_SharedFreqOnlyPwmInput.h"
+
+#define MAX_TIMER_PERIOD_ELAPSED_COUNT 2
+
+struct FreqOnlyPwmInput
+{
+    float              frequency_hz;
+    TIM_HandleTypeDef *htim;
+    float              timer_frequency_hz;
+    uint32_t           rising_edge_tim_channel;
+    uint32_t           timer_auto_reload_reg;
+    bool               first_tick;
+    uint32_t           curr_rising_edge;
+    uint32_t           prev_rising_edge;
+
+    // Elapsed tick counts the number of times which a timer overflow occurs
+    size_t elapsed_tick;
```

Comment:
```suggestion
    size_t timer_overflow_count;
```

I never understood the name of this variable and it confused me for a good while. Calling it tick makes it seem like it's counting timer ticks. If we just name it timer_overflow_count it's completely self-explanatory and you don't even need a extra comment describing what it is